src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Map.cs

```diff
@@ -0,0 +1,271 @@
+                GC.SuppressFinalize(this);
+            }
+            #endregion
+        }
+
+        /// <summary>
+        /// Class that implements an ICollection over the "keys" or "values"
+        /// </summary>
+        private sealed class MapItemCollection : ICollection
+        {
+            private readonly Map _map;
+            private readonly string _iterator;  // "keys" or "values"
+
+            public MapItemCollection(Map map, string iterator)
+            {
+                _map = map;
+                _iterator = iterator;
+
+            }
+            public int Count => _map.Count;
+
+            public bool IsSynchronized => false;
+
+            public object SyncRoot => this;
+
+            public void CopyTo(System.Array array, int index)
+            {
+                throw new NotImplementedException();
+            }
+
+            // Construct and return an enumerator.
+            public IEnumerator GetEnumerator() => new MapItemEnumerator(this);
+
+            /// <summary>
+            /// The custom enumerator used by MapItemCollection
+            /// </summary>
+            private sealed class MapItemEnumerator : IEnumerator
+            {
+
+                private readonly MapItemCollection _mapItemCollection;
+                private JSObject? _mapItemIterator;
+
+                public object? Current { get; private set; }
+
+                public MapItemEnumerator(MapItemCollection mapCollection)
+                {
+                    _mapItemCollection = mapCollection;
+                }
+
+                public bool MoveNext()
+                {
+                    if (_mapItemIterator == null)
+                        _mapItemIterator = (JSObject)_mapItemCollection._map.Invoke(_mapItemCollection._iterator);
+
+                    var done = false;
+                    using (var result = (JSObject)_mapItemIterator.Invoke("next"))
+                    {
+                        done = (bool)result.GetObjectProperty("done");
+                        if (!done)
+                            Current = result.GetObjectProperty("value");
+                        return !done;
+                    }
+                }
+
+                public void Reset()
+                {
+                    _mapItemIterator?.Dispose();
+                    _mapItemIterator = null;
+                }
+
+                #region IDisposable Support
+                private bool _disposedValue = false; // To detect redundant calls
+
+                private void Dispose(bool disposing)
+                {
+                    if (!_disposedValue)
+                    {
+                        if (disposing)
+                        {
+                            // TODO: dispose managed state (managed objects).
+                        }
+
+                        // TODO: free unmanaged resources (unmanaged objects) and override a finalizer below.
+                        // TODO: set large fields to null.
```

Comment:
What's the plan for all these TODOs?  Is there an issue tracking whatever needs to be done here?